pkg/flow/grpc-instance-variables.go

```diff
@@ -358,597 +100,65 @@ func (flow *flow) InstanceVariableParcels(req *grpc.InstanceVariableRequest, srv
 func (flow *flow) InstanceVariables(ctx context.Context, req *grpc.InstanceVariablesRequest) (*grpc.InstanceVariablesResponse, error) {
 	flow.sugar.Debugf("Handling gRPC request: %s", this())
 
-	cached, err := flow.getInstance(ctx, req.GetNamespace(), req.GetInstance())
+	instanceID, err := uuid.Parse(req.GetInstance())
 	if err != nil {
 		return nil, err
 	}
-
-	clients := flow.edb.Clients(ctx)
-
-	query := clients.VarRef.Query().Where(entvar.HasInstanceWith(entinst.ID(cached.Instance.ID)))
-
-	results, pi, err := paginate[*ent.VarRefQuery, *ent.VarRef](ctx, req.Pagination, query, variablesOrderings, variablesFilters)
+	ns, err := flow.edb.NamespaceByName(ctx, req.GetNamespace())
 	if err != nil {
 		return nil, err
 	}
-
-	resp := new(grpc.InstanceVariablesResponse)
-	resp.Namespace = cached.Namespace.Name
-	resp.Instance = cached.Instance.ID.String()
-	resp.Variables = new(grpc.Variables)
-	resp.Variables.PageInfo = pi
-
-	err = bytedata.ConvertDataForOutput(results, &resp.Variables.Results)
+	_, store, _, rollback, err := flow.beginSqlTx(ctx)
 	if err != nil {
 		return nil, err
 	}
+	defer rollback()
 
-	for i := range results {
-		vref := results[i]
+	list, err := store.RuntimeVariables().ListByInstanceID(ctx, instanceID)
+	if err != nil {
+		return nil, err
+	}
 
-		vdata, err := vref.QueryVardata().Select(entvardata.FieldCreatedAt, entvardata.FieldHash, entvardata.FieldSize, entvardata.FieldUpdatedAt).Only(ctx)
-		if err != nil {
-			return nil, err
-		}
+	resp := new(grpc.InstanceVariablesResponse)
+	resp.Namespace = ns.Name
+	resp.Instance = req.GetInstance()
+	resp.Variables = new(grpc.Variables)
+	resp.Variables.PageInfo = nil
```

Comment:
PageInfo may need a TODO. Ask Steffan if his UI is using it?

Reply:
Fixed :)